src/pages/studyView/StudyViewPageStore.ts

```diff
@@ -466,34 +470,42 @@ export class StudyViewPageStore {
-                    // create groups using data
-                    let patientToSamples:{[uniquePatientKey:string]:SampleIdentifier[]} = {};
-                    if (clinicalAttribute.patientAttribute) {
-                        patientToSamples = _.groupBy(selectedSamples, s=>s.uniquePatientKey);
-                    }
-                    const groups = quartiles.map(quartile=>{
-                        let studies;
+                    let groups;
+                    if (categorizationType === NumericalComparisonType.QUARTILES) {
+
+                        let patientToSamples: { [uniquePatientKey: string]: SampleIdentifier[] } = {};
                         if (clinicalAttribute.patientAttribute) {
-                            studies = getStudiesAttr(
-                                _.flatMapDeep(
-                                    quartile,
-                                    (d:ClinicalData)=>{
-                                        return patientToSamples[d.uniquePatientKey].map(s=>({ studyId:s.studyId, sampleId:s.sampleId }));
-                                    }
-                                )
-                            );
-                        } else {
-                            studies = getStudiesAttr(quartile.map(d=>({ studyId:d.studyId, sampleId:d.sampleId })));
+                            patientToSamples = _.groupBy(selectedSamples, s => s.uniquePatientKey);
                         }
-                        const range = [quartile[0].value, quartile[quartile.length-1].value];
-                        return {
-                            name: `${range[0]}-${range[1]}`,
-                            description: "",
-                            studies,
-                            origin: this.studyIds,
-                        };
-                    });
+
+                        const quartiles = getQuartiles(data);
+
+                        // create groups using data
+                        groups = quartiles.map(quartile => {
+                            let studies;
+                            if (clinicalAttribute.patientAttribute) {
+                                studies = getStudiesAttr(
+                                    _.flatMapDeep(
+                                        quartile,
+                                        (d: ClinicalData) => {
+                                            return patientToSamples[d.uniquePatientKey].map(s => ({ studyId: s.studyId, sampleId: s.sampleId }));
+                                        }
+                                    )
+                                );
+                            } else {
+                                studies = getStudiesAttr(quartile.map(d => ({ studyId: d.studyId, sampleId: d.sampleId })));
+                            }
+                            const range = [quartile[0].value, quartile[quartile.length - 1].value];
+                            return {
```

Comment:
not sure but i think this untyped because Promise is any.  Would be nice to have this typechecked.